addons/web/static/tests/views/fields/datetime_field.test.js

```diff
@@ -141,6 +141,42 @@ test("DatetimeField only triggers fieldChange when a day is picked and when an h
     expect.verifySteps(["onchange"]);
 });
 
+test("DatetimeField edit hour/minute and click away", async () => {
+    mockTimeZone(0);
+
+    onRpc("web_save", ({ args }) => {
+        expect(args[1].datetime).toBe("2017-02-08 08:30:00", {
+            message: "the correct value should be saved",
+        });
+    });
+
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 1,
+        arch: /* xml */ '<form><field name="datetime"/></form>',
+    });
+
+    // Open the datepicker
+    await click(".o_field_datetime input");
+    await animationFrame();
+    expect(".o_datetime_picker").toHaveCount(1);
+
+    // Manually change the time without { confirm: "enter" }
+    await click(`.o_time_picker_input:eq(0)`);
+    await animationFrame();
+    await edit("8:30");
+    await animationFrame();
```

Comment:
I would add an assert here to check that the input value hasn't been updated yet (which is a prerequisite for your test to be relevant)